apps/explorer/src/app/routes/txs/home/index.tsx

```diff
@@ -1,29 +1,93 @@
-import type { TendermintBlockchainResponse } from '../../blocks/tendermint-blockchain-response';
 import { DATA_SOURCES } from '../../../config';
+import { useCallback, useState, useMemo } from 'react';
+import { t, useFetch } from '@vegaprotocol/react-helpers';
 import { RouteTitle } from '../../../components/route-title';
 import { BlocksRefetch } from '../../../components/blocks';
-import { RenderFetched } from '../../../components/render-fetched';
-import { BlockTxsData } from '../../../components/txs';
 import { JumpToBlock } from '../../../components/jump-to-block';
-import { t, useFetch } from '@vegaprotocol/react-helpers';
+import { TxsInfiniteList } from '../../../components/txs';
+import type { ChainExplorerTxResponse } from '../../types/chain-explorer-response';
+
+interface TxsProps {
+  latestBlockHeight: string;
+  refresh: () => null;
+}
+
+interface TxsStateProps {
+  txsData: ChainExplorerTxResponse[];
+  hasMoreTxs: boolean;
+  nextPage: number;
+}
+
+const Txs = ({ latestBlockHeight }: TxsProps) => {
+  const [{ txsData, hasMoreTxs, nextPage }, setTxsState] =
+    useState<TxsStateProps>({
+      txsData: [],
+      hasMoreTxs: true,
+      nextPage: 1,
+    });
+
+  const reusedBodyParams = useMemo(
+    () => ({
+      node_url: `${DATA_SOURCES.tendermintUrl}`,
+      transaction_height: parseInt(latestBlockHeight),
+      page_size: 30,
+    }),
+    [latestBlockHeight]
+  );
 
-const Txs = () => {
   const {
-    state: { data, error, loading },
+    state: { error, loading },
     refetch,
-  } = useFetch<TendermintBlockchainResponse>(
-    `${DATA_SOURCES.tendermintUrl}/blockchain`
+  } = useFetch(
+    `${DATA_SOURCES.chainExplorerUrl}`,
+    {
+      method: 'POST',
+      body: JSON.stringify({
+        ...reusedBodyParams,
+      }),
+    },
+    false
   );
 
+  const loadTxs = useCallback(async () => {
+    setTxsState((prev) => ({
+      ...prev,
```

Comment:
question (non-blocking): is this setter here needed? looks like it doesn't do anything at the moment.

Reply:
Another good spot, more remnants of an older implementation that got refactored. Thank you!